src/Document.ts

```diff
@@ -36,25 +36,81 @@ class Document {
 	/** Indicates whether this document is a subdocument of a composing parent */
 	readonly #isSubdocument: boolean;
 
-	public constructor(schema: Schema | null, options: DocumentConstructorOptions) {
+	protected constructor(schema: Schema | null, options: DocumentConstructorOptions) {
 		const { data = {}, record, isSubdocument = false } = options;
 
 		this.#schema = schema;
-		this.#record = [];
+		this.#record = record ?? [];
 		this.#isSubdocument = isSubdocument;
-		this.transformationErrors = [];
+		this._transformationErrors = [];
 
 		Object.defineProperties(this, {
-			transformationErrors: { configurable: false, enumerable: false, writable: false },
+			_transformationErrors: { configurable: false, enumerable: false, writable: false },
 		});
 
-		if (record != null) {
-			this.#transformRecordToDocument(record);
-		}
+		this.#transformRecordToDocument();
+
 		// load the data passed to constructor into document instance
 		assignIn(this, data);
 	}
 
+	/** Create a new Subdocument instance from a record array */
+	public static createSubdocumentFromRecord(schema: Schema, record: MvRecord): Document {
+		return new Document(schema, { record, isSubdocument: true });
+	}
+
+	/** Create a new Subdocument instance from data */
+	public static createSubdocumentFromData(schema: Schema, data: GenericObject): Document {
+		return new Document(schema, { data, isSubdocument: true });
+	}
+
+	/** Create a new Document instance from a record string */
+	public static createDocumentFromRecordString(
+		schema: Schema,
+		recordString: string,
+		dbServerDelimiters: DbServerDelimiters,
+	): Document {
+		const record = Document.convertMvStringToArray(recordString, dbServerDelimiters);
+
+		return new Document(schema, { record });
+	}
+
+	/** Convert a multivalue string to an array */
+	public static convertMvStringToArray(
+		recordString: string,
+		dbServerDelimiters: DbServerDelimiters,
+	): MvRecord {
+		const { am, vm, svm } = dbServerDelimiters;
+		const record: MvRecord =
+			recordString === ''
+				? []
+				: recordString.split(am).map((attribute) => {
+						if (attribute === '') {
+							return null;
+						}
+
+						const attributeArray = attribute.split(vm);
```

Comment:
If an attribute just contains subvalues we'll never hit the below condition to further split on svm.

Reply:
I've adjusted this to test the string for inclusion of svm marks and continue if they are found.  Unit tests have been added to  test this scenario as well.